runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -475,38 +571,68 @@ TR_VectorAPIExpansion::visitNodeToBuildVectorAliases(TR::Node *node)
             if (_trace)
                traceMsg(comp(), "Can't scalarize #%d due to unsupported opcode in node %p\n",
                                  node->getSymbolReference()->getReferenceNumber(), node);
-
             _aliasTable[methodRefNum]._cantScalarize = true;
             }
+
+         createClassesForBoxing(node->getSymbolReference()->getOwningMethod(comp()), methodElementType, bitsLength);
          }
       }
    else if (opCode.isLoadAddr())
       {
       if (_trace)
-         traceMsg(comp(), "Invalidating #%d due to loadaddr node %p\n", node->getSymbolReference()->getReferenceNumber(), node);
+         traceMsg(comp(), "Invalidating7 #%d due to its adress used by loadaddr node %p\n", node->getSymbolReference()->getReferenceNumber(), node);
       invalidateSymRef(node->getSymbolReference());
       }
    else if (opCode.isArrayRef() ||
-            opCode.isLoadIndirect() ||
-            opCode.isStoreIndirect() ||
+            opCode.isLoadIndirect())
+      {
+      TR::Node *child = node->getFirstChild();
+      if (child->getOpCode().hasSymbolReference())
+         {
+         if (boxingAllowed())
+            {
+            // make it boxed since, currently, transformation pass is not recursive
+            // and we will not detect if boxing is necessary deeper in the trees
+            _aliasTable[child->getSymbolReference()->getReferenceNumber()]._vecLen = vec_len_boxed_unknown;
+            }
+         else
+            {
+            if (_trace)
+               traceMsg(comp(), "Invalidating8 #%d due to its address used by %p\n",
+                                 child->getSymbolReference()->getReferenceNumber(), node);
+               invalidateSymRef(child->getSymbolReference());
+            }
+         }
+      }
+   else if (opCode.isStoreIndirect() ||
             node->getOpCodeValue() == TR::areturn ||
             node->getOpCodeValue() == TR::aRegStore)
       {
       TR::Node *child = node->getFirstChild();
       if (child->getOpCode().hasSymbolReference())
          {
-         if (_trace)
-            traceMsg(comp(), "Invalidating #%d due to its address used by %p\n",
-                             child->getSymbolReference()->getReferenceNumber(), node);
-         invalidateSymRef(child->getSymbolReference());
+         if (!boxingAllowed())
+            {
+            if (_trace)
+               traceMsg(comp(), "Invalidating9 #%d due to its address used by %p\n",
+                                 child->getSymbolReference()->getReferenceNumber(), node);
+            invalidateSymRef(child->getSymbolReference());
+            }
          }
       }
+
+   else if (boxingAllowed() &&
+            node->getOpCodeValue() == TR::checkcast)
+      {
+      // allow this treetop if boxing
```

Comment:
Please add a comment like "do nothing here" for readability.

Reply:
done